File: generator/konfig-integration-tests/sdks/typescript-newscatcher-custom-server/typescript/search-all.test.ts
import { Newscatcher } from "./index";

describe("newscatcherapi-typescript-sdk", () => {
  const client = new Newscatcher({
    basePath: "http://127.0.0.1:4115",
    apiKey: "dummy",
  });

  it("authors", async () => {
    const getResult = await client.authors.get_all({
      authorName: "authorName_example",
      pageSize: 10,
      nerName: "nerName_example",
      publishedDatePrecision: "published_date_precision_example",
      sortBy: "sort_by_example",
    })
    console.log(getResult)
    expect(getResult).not.toBeNull();
    expect(getResult).not.toBeUndefined();
    expect(getResult.length).toBe(6);

    const postResult = await client.authors.post_all({
      author_name: "authorName_example",
      page_size: 10,
      ner_name: "nerName_example",
      published_date_precision: "published_date_precision_example",
      sort_by: "sort_by_example",
    })
    console.log(postResult)
    expect(postResult).not.toBeNull();
    expect(postResult).not.toBeUndefined(); 
    expect(getResult.length).toBe(6);
  });
  it("search", async () => {
    const getResult = await client.search.get_all({
      q: "q_example",
      searchIn: "search_in_example",
      pageSize: 10,
    })
    console.log(getResult)
    expect(getResult).not.toBeNull();
    expect(getResult).not.toBeUndefined();
    expect(getResult.length).toBe(6);

    const postResult = await client.search.post_all({
      q: "q_example",
      page_size: 10,
      sort_by: "sort_by_example",
    })
    console.log(postResult)
    expect(postResult).not.toBeNull();
    expect(postResult).not.toBeUndefined(); 
    expect(getResult.length).toBe(6);
  });
  it("search similar", async () => {
    const getResult = await client.searchSimilar.get_all({
      q: "q_example",
      includeSimilarDocuments: true,
      pageSize: 10,
      nerName: "nerName_example",
      publishedDatePrecision: "published_date_precision_example",
      sortBy: "sort_by_example",
    })
    console.log(getResult)
    expect(getResult).not.toBeNull();
    expect(getResult).not.toBeUndefined();
    expect(getResult.length).toBe(6);

    const postResult = await client.searchSimilar.post_all({
      q: "q_example",
      page_size: 10,
      include_similar_documents: true,
      ner_name: "nerName_example",
      published_date_precision: "published_date_precision_example",
      sort_by: "sort_by_example",
    })
    console.log(postResult)
    expect(postResult).not.toBeNull();
    expect(postResult).not.toBeUndefined(); 
    expect(getResult.length).toBe(6);
  });
  it("latest headlines", async () => {
    const getResult = await client.latestHeadlines.get_all({
      lang: "lang_example",
      pageSize: 10,
    })
    console.log(getResult)
    expect(getResult).not.toBeNull();
    expect(getResult).not.toBeUndefined();
    expect(getResult.length).toBe(6);

    const postResult = await client.latestHeadlines.post_all({
      lang: "lang_example",
      page_size: 10,
    })
    console.log(postResult)
    expect(postResult).not.toBeNull();
    expect(postResult).not.toBeUndefined(); 
    expect(getResult.length).toBe(6);
  });
  it("search by link", async () => {
    const getResult = await client.searchLink.get_all({
      ids: "ids_example",
      pageSize: 10,
    })
    console.log(getResult)
    expect(getResult).not.toBeNull();
    expect(getResult).not.toBeUndefined();
    expect(getResult.length).toBe(6);

    const postResult = await client.searchLink.post_all({
      ids: "ids_example",
      page_size: 10,
    })
    console.log(postResult)
    expect(postResult).not.toBeNull();
    expect(postResult).not.toBeUndefined(); 
    expect(getResult.length).toBe(6);
  });
});

File: generator/konfig-integration-tests/sdks/java-newscatcher-custom-server/java/src/test/java/com/konfigthis/newscatcherapi/client/SearchAllTest.java
package com.konfigthis.newscatcherapi.client;

import com.konfigthis.newscatcherapi.client.model.ArticlesPropertyInner;
import com.konfigthis.newscatcherapi.client.model.ArticlesPropertyInner1;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

import java.util.List;

public class SearchAllTest {
    final String MOCK_SERVER_URL = "http://localhost:4085";

    @Test
    public void authorsTest() {
        Configuration configuration = new Configuration();
        configuration.host = MOCK_SERVER_URL;
        configuration.apiKey = "API KEY";
        Newscatcher client = new Newscatcher(configuration);

        List<ArticlesPropertyInner> getResult = client.authors.get("author_name_example")
            .publishedDatePrecision("published_date_precision_example")
            .sortBy("sort_by_example")
            .nerName("ner_name_example")
            .executeSearchAll();
        Assertions.assertNotNull(getResult);
        Assertions.assertEquals(6, getResult.size());

        List<ArticlesPropertyInner> postResult = client.authors.post("author_name_example")
                .publishedDatePrecision("published_date_precision_example")
                .sortBy("sort_by_example")
                .nerName("ner_name_example")
                .executeSearchAll();
        Assertions.assertNotNull(postResult);
        Assertions.assertEquals(6, postResult.size());
    }

    @Test
    public void latestHeadlinesTest() {
        Configuration configuration = new Configuration();
        configuration.host = MOCK_SERVER_URL;
        configuration.apiKey = "API KEY";
        Newscatcher client = new Newscatcher(configuration);

        List<Object> getResult = client.latestHeadlines.get()
                .when("when_example")
                .executeSearchAll();
        Assertions.assertNotNull(getResult);
        Assertions.assertEquals(6, getResult.size());

        List<Object> postResult = client.latestHeadlines.post()
                .when("when_example")
                .executeSearchAll();
        Assertions.assertNotNull(postResult);
        Assertions.assertEquals(6, postResult.size());
    }

    @Test
    public void searchTest() {
        Configuration configuration = new Configuration();
        configuration.host = MOCK_SERVER_URL;
        configuration.apiKey = "API KEY";
        Newscatcher client = new Newscatcher(configuration);

        List<Object> getResult = client.search.get("q_example")
            .searchIn("search_in_example")
            .publishedDatePrecision("published_date_precision_example")
            .sortBy("sort_by_example")
            .executeSearchAll();
        Assertions.assertNotNull(getResult);
        Assertions.assertEquals(6, getResult.size());

        List<Object> postResult = client.search.post("q_example")
                .searchIn("search_in_example")
                .publishedDatePrecision("published_date_precision_example")
                .sortBy("sort_by_example")
                .executeSearchAll();
        Assertions.assertNotNull(postResult);
        Assertions.assertEquals(6, postResult.size());
    }

    @Test
    public void searchLinkTest() {
        Configuration configuration = new Configuration();
        configuration.host = MOCK_SERVER_URL;
        configuration.apiKey = "API KEY";
        Newscatcher client = new Newscatcher(configuration);

        List<Object> getResult = client.searchLink.get().executeSearchAll();
        Assertions.assertNotNull(getResult);
        Assertions.assertEquals(6, getResult.size());

        List<Object> postResult = client.searchLink.post().executeSearchAll();
        Assertions.assertNotNull(postResult);
        Assertions.assertEquals(6, postResult.size());
    }

    @Test
    public void searchSimilarTest() {
        Configuration configuration = new Configuration();
        configuration.host = MOCK_SERVER_URL;
        configuration.apiKey = "API KEY";
        Newscatcher client = new Newscatcher(configuration);

        List<ArticlesPropertyInner1> getResult = client.searchSimilar.get("q_example")
                .searchIn("search_in_example")
                .publishedDatePrecision("published_date_precision_example")
                .sortBy("sort_by_example")
                .includeSimilarDocuments(true)
                .similarDocumentsNumber(100)
                .similarDocumentsFields("similar_documents_fields_example")
                .nerName("ner_name_example")
                .executeSearchAll();
        Assertions.assertNotNull(getResult);
        Assertions.assertEquals(6, getResult.size());

        List<ArticlesPropertyInner1> postResult = client.searchSimilar.post("q_example")
                .searchIn("search_in_example")
                .publishedDatePrecision("published_date_precision_example")
                .sortBy("sort_by_example")
                .includeSimilarDocuments(true)
                .similarDocumentsNumber(100)
                .similarDocumentsFields("similar_documents_fields_example")
                .nerName("ner_name_example")
                .executeSearchAll();
        Assertions.assertNotNull(postResult);
        Assertions.assertEquals(6, postResult.size());
    }
}
